aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java

```diff
@@ -113,26 +115,44 @@ boolean keepTokenRefreshed() {
         OAuth2Properties.TOKEN_REFRESH_ENABLED_DEFAULT);
   }
 
+  private ScheduledExecutorService tokenRefreshExecutor() {
```

Comment:
Shouldn't this be static?

Reply:
unfortunately it can't be static, because it accesses non-static stuff (`keepTokenRefreshed()`)

Reply:
I think it can be if we move the `if (!keepTokenRefreshed())` out of the method and go back to the ternary operator we had . everything else looks static.

Reply:
right, we can make the method static if we move `keepTokenRefreshed()` out, but I'd like to keep the logic inside this method (similar to how we have it in https://github.com/apache/iceberg/blob/fa6fa237ebaa4ac6c4b0c40b117aff22466d2b17/core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java#L485-L487)